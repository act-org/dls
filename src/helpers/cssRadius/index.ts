/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { StandardCSSProperties } from '@mui/system';

import px from '~/helpers/px';

type BorderRadius = Pick<
  StandardCSSProperties,
  | 'borderBottomLeftRadius'
  | 'borderBottomRightRadius'
  | 'borderRadius'
  | 'borderTopLeftRadius'
  | 'borderTopRightRadius'
>;

export const cssRadius = (value: number): BorderRadius => ({
  borderBottomLeftRadius: px(value),
  borderBottomRightRadius: px(value),
  borderRadius: px(value),
  borderTopLeftRadius: px(value),
  borderTopRightRadius: px(value),
});

export default cssRadius;
