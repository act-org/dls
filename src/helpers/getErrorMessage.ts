/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { get } from 'lodash';

export const getErrorMessage = (err: unknown): string => {
  let errorMessage = '';
  const graphQlErrorMessage = get(err, 'networkError.result.errors[0].message');
  const standardErrorMessage = get(err, 'message');

  if (graphQlErrorMessage) {
    errorMessage = graphQlErrorMessage;
  } else if (standardErrorMessage) {
    errorMessage = standardErrorMessage;
  }

  return errorMessage.trim();
};

export default getErrorMessage;
