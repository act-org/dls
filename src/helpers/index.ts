/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

export * from './cssRadius';
export * from './getErrorMessage';
export * from './makeShadow';
export * from './mergeClasses';
export * from './oxfordCommaJoin';
export * from './px';
export * from './pxToNumber';
export * from './search';
export * from './sort';
export * from './styled';
export * from './types';
