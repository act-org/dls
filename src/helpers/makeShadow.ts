/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import px from '@actinc/dls/helpers/px';

export const makeShadow = (
  xOffset: number,
  yOffset: number,
  blurRadius: number | undefined,
  spreadRadius: number | undefined,
  color: string,
  inset = false,
): string => {
  return `${inset ? 'inset ' : ''}${px(xOffset)} ${px(yOffset)} ${`${px(
    blurRadius,
  )} `}${`${px(spreadRadius)} `}${color}`;
};

export default makeShadow;
