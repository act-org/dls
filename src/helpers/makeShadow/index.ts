/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import isNumber from 'lodash/isNumber';

import px from '~/helpers/px';

export const makeShadow = (
  xOffset: number,
  yOffset: number,
  blurRadius: number | undefined,
  spreadRadius: number | undefined,
  color: string,
  inset = false,
): string => {
  let str = '';

  if (inset) {
    str = `${str} inset`;
  }

  if (isNumber(xOffset)) {
    str = `${str} ${px(xOffset)}`;
  }

  if (isNumber(yOffset)) {
    str = `${str} ${px(yOffset)}`;
  }

  if (isNumber(blurRadius)) {
    str = `${str} ${px(blurRadius)}`;
  }

  if (isNumber(spreadRadius)) {
    str = `${str} ${px(spreadRadius)}`;
  }

  if (color) {
    str = `${str} ${color}`;
  }

  return str.trim();
};

export default makeShadow;
