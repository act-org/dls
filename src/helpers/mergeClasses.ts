/**
 * Copyright (c) Encoura, LLC and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import clsx from 'clsx';
import mergeWith from 'lodash/mergeWith';

type Classes = any;

export const mergeClasses = (classes1: Classes, classes2: Classes): Classes =>
  mergeWith(classes1 || {}, classes2 || {}, (className1, className2): string =>
    clsx(className1, className2),
  );

export default mergeClasses;
