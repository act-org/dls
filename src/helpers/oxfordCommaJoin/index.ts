/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export const oxfordCommaJoin = (input: string[]): string => {
  const clone = [...input];

  if (input.length <= 2) {
    return clone.join(' and ');
  }

  const last = clone.pop();
  return `${clone.join(', ')}, and ${last}`;
};

export default oxfordCommaJoin;
