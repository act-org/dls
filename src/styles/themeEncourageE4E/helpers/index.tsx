/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export type ColorIndex =
  | 50
  | 100
  | 200
  | 300
  | 400
  | 500
  | 600
  | 700
  | 800
  | 900
  | string;

interface IBrightnessMap {
  brightness: number;
  colorIndex: ColorIndex;
}

/*
 * Do not modify these values unless UX give us new values,
 * they are used to generate our Figma palette
 */
const brightnessCoefficient: IBrightnessMap[] = [
  {
    brightness: 0.9,
    colorIndex: 50,
  },
  {
    brightness: 0.69,
    colorIndex: 100,
  },
  {
    brightness: 0.54,
    colorIndex: 200,
  },
  {
    brightness: 0.33,
    colorIndex: 300,
  },
  {
    brightness: 0.2,
    colorIndex: 400,
  },
  {
    brightness: 0,
    colorIndex: 500,
  },
  {
    brightness: -0.09,
    colorIndex: 600,
  },
  {
    brightness: -0.29,
    colorIndex: 700,
  },
  {
    brightness: -0.45,
    colorIndex: 800,
  },
  {
    brightness: -0.58,
    colorIndex: 900,
  },
];

export const hexToRGB = (color: string): number[] | undefined => {
  let r = '';
  let g = '';
  let b = '';

  if ((color.length !== 4 && color.length !== 7) || color[0] !== '#')
    return undefined;

  if (color.length === 4) {
    r = `0x${color[1]}${color[1]}`;
    g = `0x${color[2]}${color[2]}`;
    b = `0x${color[3]}${color[3]}`;
  } else if (color.length === 7) {
    r = `0x${color[1]}${color[2]}`;
    g = `0x${color[3]}${color[4]}`;
    b = `0x${color[5]}${color[6]}`;
  }

  return [Number(r), Number(g), Number(b)];
};

export const RGBToHex = (r: number, g: number, b: number): string => {
  let R = r.toString(16);
  let G = g.toString(16);
  let B = b.toString(16);

  if (R.length === 1) R = `0${R}`;
  if (G.length === 1) G = `0${G}`;
  if (B.length === 1) B = `0${B}`;

  return `#${R.slice(0, 2)}${G.slice(0, 2)}${B.slice(0, 2)}`;
};

export const lightenRGB = (
  color: string,
  percentage: number,
): string | undefined => {
  const RGBColor = hexToRGB(color);

  if (!RGBColor) return undefined;

  if (percentage === 0) return color;

  if (percentage > 0) {
    return RGBToHex(
      Math.round(RGBColor[0] + ((255 - RGBColor[0]) * percentage) / 100),
      Math.round(RGBColor[1] + ((255 - RGBColor[1]) * percentage) / 100),
      Math.round(RGBColor[2] + ((255 - RGBColor[2]) * percentage) / 100),
    );
  }

  return RGBToHex(
    Math.round(RGBColor[0] * (1 + percentage / 100)),
    Math.round(RGBColor[1] * (1 + percentage / 100)),
    Math.round(RGBColor[2] * (1 + percentage / 100)),
  );
};

export const figmaColorGenerator = (
  color: string,
): Record<ColorIndex, string> => {
  const colorPalette = {} as unknown as Record<ColorIndex, string>;
  brightnessCoefficient.forEach(({ brightness, colorIndex }) => {
    const generatedColor = lightenRGB(color, brightness * 100);
    if (generatedColor) colorPalette[colorIndex] = generatedColor;
  });
  return colorPalette;
};

export const generatePaletteColors = <
  T extends { colorName: U; colorHEX: string },
  U extends string,
>(
  colors: T[],
): Record<U, Record<ColorIndex, string>> => {
  const colorPalette = {} as unknown as Record<U, Record<ColorIndex, string>>;
  colors.forEach(color => {
    colorPalette[color.colorName] = figmaColorGenerator(color.colorHEX);
  });
  return colorPalette;
};
