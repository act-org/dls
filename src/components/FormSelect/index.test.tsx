/**
 * Copyright (c) Encoura, LLC and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { standard } from '~/helpers/test';

import { FormSelect } from '.';

describe('FormSelect', () => {
  const Component = (
    <FormSelect
      label="Select Label"
      options={[
        {
          label: 'One',
          value: 'ONE',
        },
        {
          label: 'Two',
          value: 'TWO',
        },
        {
          label: 'Three',
          value: 'THREE',
        },
      ]}
      placeholder="Placeholder"
      value="ONE"
      variant="outlined"
    />
  );
  standard(Component);
});
