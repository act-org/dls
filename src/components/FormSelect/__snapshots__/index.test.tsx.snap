// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelect ACT theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-yblnas-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      <div
        class="label-split"
      >
        <span
          class="label-help"
        >
          Select Label
        </span>
      </div>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-prxojz-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-9t6dra-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="button"
        tabindex="0"
      >
        One
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="ONE"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-115bljm-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-hdw1oc"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`FormSelect ACT_ET theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-16xra3p-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      <div
        class="label-split"
      >
        <span
          class="label-help"
        >
          Select Label
        </span>
      </div>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-11956l7-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-1jfpuyl-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="button"
        tabindex="0"
      >
        One
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="ONE"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-9q8pxw-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-hdw1oc"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`FormSelect ENCOURA_DATALAB theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-742zqy-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      <div
        class="label-split"
      >
        <span
          class="label-help"
        >
          Select Label
        </span>
      </div>
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1uoibu0-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-p8sous-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="button"
        tabindex="0"
      >
        One
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="ONE"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-1cg639m-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ChevronDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-hdw1oc"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
